include/cartesian_geom/autopoint.h

```diff
@@ -0,0 +1,311 @@
+#ifndef AB9D4FD9_C418_44AC_8276_CC42540EF027
```

Comment:
Would it make sense here to have a differentiate() function that computes the gradient of the point?

Reply:
autopoint on its own doesn't differentiate as there is no function. 
Autopoint functor will have to differentiate () public interface.

Reply:
ok